### PR TITLE
fix: Delete steps in editor (replace with null) doesn't propagate to …

### LINKED ISSUE #744: Delete steps in editor (replace with null) doesn't propagate to the canvas
### Describe the Bug

When users replace the end step with "null" - which should delete the step, syncing the changes doesn't get reflected in the canvas.

### Steps to Reproduce the Bug or Issue

1. create integration with start and end steps
2. replace the end step with "null"
3. sync the changes
4. see the end step doesn't get removed from the canvas

### Screenshots or Videos

Peek 2023-07-03 09-16

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [latest]

### PR DESCRIPTION
**Pull request by igarashitm**
…the canvas

Fixes: #744

### CODE REVIEW

File: kamelet-support/src/test/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletBindingStepParserServiceTest.java [hunk 1]
```diff
@@ -128,4 +128,58 @@ void appliesToInvalid(String resourcePath) throws IOException {
...
+    }
+
+    @Test
+    void parseNullSourceNullSink() throws IOException {
+        String input = new String(
+                Objects.requireNonNull(
+                        this.getClass().getResourceAsStream("null-source-null-sink.binding.yaml"))
+                .readAllBytes(), StandardCharsets.UTF_8);
+        StepParserService.ParseResult<Step> parsed = dslSpecification.getStepParserService().deepParse(input);
+        assertThat(parsed.getSteps()).hasSize(0);
```
Comment (lordrip) [inline kamelet-support/src/test/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletBindingStepParserServiceTest.java:179]: Is this a big deal? :thinking: 

https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=746&id=KaotoIO_kaoto-backend&open=AYkdVe_-Ber657K2ULFM

**Review by lordrip — APPROVED** [review]
Tried:

1. Adding source, step, and sink and then delete the sink from the source code
2. Adding source, step, and sink and then delete the sink from the canvas
3. Adding source, step, and sink and remove the sink line

so far the sink property returns as `null` :+1:

File: kamelet-support/src/test/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletBindingStepParserServiceTest.java (hunk 1 above)
Comment (Delawen) [reply kamelet-support/src/test/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletBindingStepParserServiceTest.java:179]: It's supposed to be more optimized, I guess.

Comment (Delawen) [reply kamelet-support/src/test/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletBindingStepParserServiceTest.java:179]: I guess it is worth adding that to the PR.

**Review by Delawen — CHANGES_REQUESTED** [review]
Just

File: kamelet-support/src/test/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletBindingStepParserServiceTest.java (hunk 1 above)
Comment (igarashitm) [reply kamelet-support/src/test/java/io/kaoto/backend/api/service/step/parser/kamelet/KameletBindingStepParserServiceTest.java]: done :+1:

**Review by Delawen — APPROVED** [review]

**Pull request merged**